src/cashlib/cashlib.cpp

```diff
@@ -1006,6 +1007,79 @@ extern "C" JNIEXPORT jbyteArray JNICALL Java_bitcoinunlimited_libbitcoincash_Has
     return bArray;
 }
 
+class CDecodablePartialMerkleTree : public CPartialMerkleTree
+{
+public:
+    std::vector<uint256> &accessHashes() { return vHash; }
+    CDecodablePartialMerkleTree(unsigned int ntx, char *bitField, int bitFieldLen)
```

Comment:
why  std::vector<uint256> &  instead of just std::vector<uint256> ?  there is no extra copy memallocation since c++11 on  return value of a vector.

Reply:
If its moved then it disappears out of the object which is not what we want.  What I want to do is load this array with values so I can later call a method on the object.

The larger context of what is going on here is that it looks to me like the parent class has suffered from being "de-featured"... Bitcoind never needs to actually decode a merkle proof, it only encodes them.  So even though the key function to do so was left in the code, the ability actually do so disappeared because there's no way to construct a CPartialMerkleTree with a bunch of bytes that are meant to be a merkle proof.  You can only construct one from a block -- that is encode a merkle proof.  This derived class provides access to the parent class internals, allowing a CPartialMerkleTree to be loaded with a merkle proof for subsequent validation.

The goal here is to touch the main bitcoin code as little as possible which is why I didn't just add ctors to construct the CPartialMerkleTree.  But if in the future anything other than cashlib needs this functionality, we should move my derived classes' functions into the parent.

Reply:
ok great